src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -245,6 +247,14 @@ protected void shadowUpdated(Map<String, Object> desired, Integer version) {
         boolean cancelDeployment = DESIRED_STATUS_CANCELED.equals(desired.get(DESIRED_STATUS_KEY));
 
         synchronized (ShadowDeploymentListener.class) {
```

Comment:
the code in the method isn't very nullsafe, can you go over all of it and add null checks when reading the fields?